File: ui/js-lib/file-storage-zome-client.js
export class FileStorageZomeClient {

    constructor(callZome) {
        this.callZome = callZome;
    }

    storeFile(file) {
        console.log('TODO: Store file ' + JSON.stringify(file));
        return "ab8629cd989324829f9aa98";
    }

    getFile(manifestAddress) {
        console.log('TODO: Get file ' + JSON.stringify(manifestAddress));

        // We'll just return some mock data here. A red dot png.
        const byteCharacters = atob('iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg==');
        const byteNumbers = new Array(byteCharacters.length);
        for (let i = 0; i < byteCharacters.length; i++) {
            byteNumbers[i] = byteCharacters.charCodeAt(i);
        }
        const byteArray = new Uint8Array(byteNumbers);

        return byteArray;
    }

}

File: ui/src/file_storage_zome_client.rs
use js_sys::Function;
use stdweb::web::TypedArray;
use stdweb::unstable::TryInto;

#[allow(dead_code)]
pub struct FileStorageZomeClient {
    call_zome: Function,
}

impl FileStorageZomeClient {
    pub fn new(call_zome: Function) -> Self {
        js! { console.log("FileStorageZomeClient initialized") };
        Self { call_zome }
    }

    pub fn store_file(&self, file: stdweb::Value) -> String {
        js! { console.log("TODO: Store file " + JSON.stringify(@{file})) };
        "ab8629cd989324829f9aa98".into()
    }

    pub fn get_file(&self, manifest_address: &str) -> TypedArray<u8> {
        js! { console.log("TODO: Get file " + JSON.stringify(@{manifest_address})) };

        // We'll just return some mock data here. A red dot png.
        let byte_characters = base64::decode(
            "iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg=="
        ).unwrap();

        byte_characters[..].into()
    }
}
